scipy/stats/_continuous_distns.py

```diff
@@ -8545,7 +8545,7 @@ def _updated_ctor_param(self):
 
 
 # Collect names of classes and objects in this module.
-pairs = list(globals().items())
+pairs = list(globals().copy().items())
```

Comment:
This seems to be the suggestion from @andyfaff in the matching issue.

I do note that we don't actually appear to have a reliable reproducer here and that the problem appears to only show up in the context of certain parallelization schemes?

Reply:
Yes, it's hard to reproduce. It's been reported before, but there's no test case that can make it happen everytime. Hopefully the copy is atomic enough.